### PR TITLE
Add -AsUTC to the Get-Date cmdlet

### PR DESCRIPTION
**Pull request by brendandburns**
# PR Summary
Adds a `-AsUTC` flag to convert a date to UTC before formatting.

## PR Context

Requested in #11410

## PR Checklist

- [X] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [X] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [X] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [X] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [X] None
- **User-facing changes**
    - [X] Not Applicable
- **Testing - New and feature**
    - [X] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [X] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.

### CODE REVIEW

File: src/Microsoft.PowerShell.Commands.Utility/commands/utility/GetDateCommand.cs [hunk 1]
```diff
@@ -217,6 +217,19 @@ public int Millisecond
...
+        /// </summary>
+        [Parameter(ParameterSetName = "net")]
+        public SwitchParameter AsUTC
+        {
+            get { return _asUTC; }
+
+            set { _asUTC = value; }
+        }
+
+        private SwitchParameter _asUTC = false;
```
Comment (iSazonov) [inline src/Microsoft.PowerShell.Commands.Utility/commands/utility/GetDateCommand.cs]: Please don't use private fields.
`public SwitchParameter AsUTC { get; set; }`

File: test/powershell/Modules/Microsoft.PowerShell.Utility/Get-Date.Tests.ps1 [hunk 2]
```diff
@@ -24,6 +24,11 @@ Describe "Get-Date DRT Unit Tests" -Tags "CI" {
         Get-date -Date:"Jan 1, 2020"  -Format:"MMM-dd-yy" | Should -Be "Jan-01-20"
     }
 
+    It "using -AsUTC produces the correct output" {
+        Get-date -Date:"2020-01-01T00:00:00" -Format:"MMM-dd-yy hh:mm" | Should -Be "Jan-01-20 12:00"
+        Get-date -Date:"2020-01-01T00:00:00" -Format:"MMM-dd-yy hh:mm" -AsUTC | Should -Be "Jan-01-20 08:00"
```
Comment (iSazonov) [inline test/powershell/Modules/Microsoft.PowerShell.Utility/Get-Date.Tests.ps1]: Perhaps follow will be more reliable test:
```powershell
(Get-date -Date:"2020-01-01T00:00:00").Kind | Should -Be [DateTimeKind]::Local
(Get-date -Date:"2020-01-01T00:00:00" -AsUTC).Kind | Should -Be [DateTimeKind]::Utc
```

**Comment by iSazonov** [comment]
@brendandburns Please don't edit/trim "PR Checklist".

File: src/Microsoft.PowerShell.Commands.Utility/commands/utility/GetDateCommand.cs (hunk 1 above)
Comment (brendandburns) [reply src/Microsoft.PowerShell.Commands.Utility/commands/utility/GetDateCommand.cs]: done.
(fwiw, I was trying to replicate how other fields in this Cmdlet are handled with public/private fields)

File: test/powershell/Modules/Microsoft.PowerShell.Utility/Get-Date.Tests.ps1 (hunk 2 above)
Comment (brendandburns) [reply test/powershell/Modules/Microsoft.PowerShell.Utility/Get-Date.Tests.ps1]: Done
(I had to modify the test a little from what you wrote, but I think it captures the same thing)

**Comment by brendandburns** [comment]
Comments addressed, apologies for clipping the checklist, still learning the ropes around here :)

Please take another look.
Thanks!

File: src/Microsoft.PowerShell.Commands.Utility/commands/utility/GetDateCommand.cs (hunk 1 above)
Comment (brendandburns) [reply src/Microsoft.PowerShell.Commands.Utility/commands/utility/GetDateCommand.cs]: fwiw, the Codacy PR tool disagrees with you :)

https://app.codacy.com/manual/TravisEz13/PowerShell_PowerShell/pullRequest?prid=4837266

File: src/Microsoft.PowerShell.Commands.Utility/commands/utility/GetDateCommand.cs [hunk 3]
```diff
@@ -217,6 +217,11 @@ public int Millisecond
         [ArgumentCompletions("FileDate", "FileDateUniversal", "FileDateTime", "FileDateTimeUniversal")]
         public string Format { get; set; }
 
+        /// <summary>
+        /// Convert date to UTC before formatting.
```
Comment (iSazonov) [inline src/Microsoft.PowerShell.Commands.Utility/commands/utility/GetDateCommand.cs]: The comment should be started with "Gets or sets ..."

**Review by iSazonov — APPROVED** [review]
LGTM with one comment

File: src/Microsoft.PowerShell.Commands.Utility/commands/utility/GetDateCommand.cs (hunk 3 above)
Comment (iSazonov) [reply src/Microsoft.PowerShell.Commands.Utility/commands/utility/GetDateCommand.cs]: @brendandburns Please address the request.

**Comment by anmenaga** [comment]
@brendandburns Thank you!

**Pull request merged**